src/main/java/org/apache/sling/servlets/post/impl/SlingPostServlet.java

```diff
@@ -234,6 +235,11 @@ protected void doPost(final SlingHttpServletRequest request,
             } catch (ResourceNotFoundException rnfe) {
                 htmlResponse.setStatus(HttpServletResponse.SC_NOT_FOUND,
                     rnfe.getMessage());
+            } catch (final PersistenceException pe) {
+                log.warn("PersistenceException while handling POST "
+                  + request.getResource().getPath() + " with "
+                  + operation.getClass().getName(), pe);
+                htmlResponse.setStatus(HttpServletResponse.SC_METHOD_NOT_ALLOWED,"invalid POST request");
```

Comment:
How are we sure (or just mostly certain) that a persistence exception is a 405 and not a 500? while agreeing on the general topic of that change, we should not revert the fake info and say everything is an ACL issue. Is this what you were mentioning at the beginning of the PR? Do you think we could add tests mocking real world issue?

Reply:
There are many different types of errors (not having write permissions on content is one of them), and the various types of operations do not clearly distinguish between them, but always a PersistenceException. Also in quite some cases the JCR API is not really granular and just throws RepositoryExceptions if something is not working as expected. 

So I tried to be generic and create a kind of universal "client side error". If someone needs a more granular handling and support more types of client errors. 

What do you mean with

> Do you think we could add tests mocking real world issue

Do you see real world issues which would require a different handling?

Reply:
> There are many different types of errors (not having write permissions on content is one of them), and the various types of operations do not clearly distinguish between them, but always a PersistenceException. Also in quite some cases the JCR API is not really granular and just throws RepositoryExceptions if something is not working as expected.
> 

Maybe this would be a good time to do the extra work to identify those places and throw more specific exceptions?

Reply:
Regarding your request to make the exception handling in the Sling Post Servlet more specific: I don't plan to do that right now, because I don't have that requirement. And without an understanding what I want to achieve with it, the implementation will most likely be not that, what someone else might need (or not).